ofdb-core/src/usecases/send_update_reminders.rs

```diff
@@ -0,0 +1,267 @@
+use super::prelude::{Email as EmailAddress, *};
+use crate::gateways::email::EmailGateway;
+use time::Duration;
+
+pub fn send_update_reminders<R, G, F>(
+    repo: &R,
+    email_gateway: &G,
+    formatter: &F,
+    target_contact: TargetContact,
+    unchanged_since: Timestamp,
+    reminder_interval: Duration,
+) -> Result<()>
+where
+    R: PlaceRepo + SubscriptionRepo + ReminderRepo + UserRepo,
+    G: EmailGateway,
+    F: EmailReminderFormatter,
+{
+    let outdated_places = find_places_not_updated_since(repo, unchanged_since)?;
+    let unsent_reminders =
+        find_unsent_reminders(repo, outdated_places, target_contact, reminder_interval)?;
+    let unsent_emails = create_emails(formatter, unsent_reminders);
+    send_emails(repo, email_gateway, unsent_emails);
+    Ok(())
+}
+
+#[derive(Debug, Clone, Copy)]
+pub enum TargetContact {
+    Owner,
+    Scout,
+}
+
+fn find_places_not_updated_since<R>(
+    place_repo: &R,
+    unchanged_since: Timestamp,
+) -> Result<Vec<Place>>
+where
+    R: PlaceRepo,
+{
+    let places = place_repo
+        .find_places_not_updated_since(unchanged_since)?
+        .into_iter()
+        .map(|(place, _)| place)
+        .collect();
+    Ok(places)
+}
+
+fn find_unsent_reminders<R>(
+    repo: &R,
+    outdated_places: Vec<Place>,
+    target_contact: TargetContact,
+    reminder_interval: Duration,
+) -> Result<Vec<Reminder>>
+where
+    R: SubscriptionRepo + ReminderRepo + UserRepo,
+{
+    let mut reminders = vec![];
+    match target_contact {
```

Comment:
Seeing the differences among the duplicated code is difficult. This could be improved by extracting the iteration over the places into the outer level and then collecting either 1 or multiple recipients for each place.